### DISCUSSION TITLE
Symfony 3.4 occasionally slow

### DISCUSSION THREAD

**Discussion by jarmo** [symfony/symfony#43412] [Q&A]
We have a legacy Symfony 3.4 application running in production. We have not written it ourselves, but need to solve a performance problem. Namely, we see that sometimes very simple requests take ~5+ seconds and then the same requests take few hundreds milliseconds later when trying again.

After some thorough log statements here and there we've narrowed it down by running `php bin/console debug:event-dispatcher kernel.request` that the problem is in `TraceableFirewallListener` and in the end `Symfony\Component\Security\Http\Firewall\ContextListener` which calls `Symfony\Bridge\Doctrine\Security\User\EntityUserProvider` `supportsClass` function which takes 5+ seconds. We have not gotten any deeper as of this moment, but since it is really low-level code already then I decided to ask help in here in the hopes of someone having had a similar problem in the past and knows where to point.

Stack is running on FreeBSD 12.1 using Nginx + PHP-FPM and PHP version is 7.1.

**Comment by ruudk** [comment]
Try to use Blackfire.io to profile and see what's going on.

**Comment by jarmo** [comment]
@ruudk thank you for your reply. Do you have a great guide for using Blackfire for developer who has zero experience with that tool before? Since it is a production server, is there a chance that something will blow up during installation/after installation and production environment will become unusable? :)

**Reply by ruudk** [reply]
I don't have that. Check the docs on blackfire.io. But it will require you to install an extension that you prob want to test before doing that on production.

Maybe you should check and confirm that you have proper production settings for Symfony and Doctrine. For example run `php bin/console doctrine:ensure-production-settings`.

**Comment by jarmo** [comment]
@ruudk thank you again for your reply.

The problem with testing out blackfire.io in some non-production environment is complicated because there are no other environments as of this time. As I said, it is a legacy application.

Running `php bin/console doctrine:ensure-production-settings` gives this output:
```
[ERROR] Proxy Classes are always regenerating.
```

I expect this is not good, but can it be the source of the problem?

**Reply by jarmo** [reply]
Okay false alarm, because I ran that command again with `--no-debug` and everything is in order and I know that `kernel.debug` is false for that environment in production.

**Reply by ruudk** [reply]
Not sure, but good to change anyway.

```yaml
doctrine:
  orm:
    auto_generate_proxy_classes: '%kernel.debug%'
```

This will make sure you have the correct setting in production.

**Reply by jarmo** [reply]
I've already verified that `$kernel->isDebug()` returns false during http requests so that should not be an issue.
